### PR TITLE
Add Parser of command /usr/bin/vdo status

### PR DESCRIPTION
**Pull request by xhe123**
Add Parser and Combiner for command '/usr/bin/vdo status'
SPEC JIRA: https://projects.engineering.redhat.com/browse/CEECBA-2476

This combiner includes GETTING these elements:
- vdo slab size
- vdo volumns
- vdo data blocks used
- vdo logical blocks used
- vdo overhead blocks used
- vdo physical blocks
- vdo savings ratio
- vdo physical used pct
- vdo logical free savdings ratio pct
- vdo physical free

This parser includes GETTING following data of output '/usr/bin/vdo status'
- Kernel module
- Configuration
- VDOs
- VDO status 

Signed-off-by: Xiaonan He <xhe@redhat.com>

### CODE REVIEW

**Comment by xhe123** [comment]
Parser and combiner code coverage are 100%

Here is the code coverage for parser
```
(.python2) [xiaonan@dhcp-136-18 insights-core]$ pytest -svk vdo_status --cov=insights/parsers/ |grep vdo_status
insights/combiners/tests/test_vdo_status_calculation.py::test_values PASSED
insights/combiners/tests/test_vdo_status_calculation.py::test_documentation PASSED
insights/combiners/tests/test_vdo_status_calculation.py::test_exp PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_simple PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_empty PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_full PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_documentation PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_exp PASSED
insights/parsers/tests/test_vdo_status.py::test_vdo_status_exp2 PASSED
insights/parsers/vdo_status.py         
```                  7      0      2      0   100%
Here is the code coverage for combiner
```
(.python2) [xiaonan@dhcp-136-18 insights-core]$ pytest -svk vdo_status_calculation --cov=insights/combiners |grep vdo_status_calculation
insights/combiners/tests/test_vdo_status_calculation.py::test_values PASSED
insights/combiners/tests/test_vdo_status_calculation.py::test_documentation PASSED
insights/combiners/tests/test_vdo_status_calculation.py::test_exp PASSED
insights/combiners/vdo_status_calculation.py          66      0      4      0   100%
```

File: insights/combiners/vdo_status_calculation.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        self.slab_size = self.data['VDOs'][vol]['Slab size']
+        self.physical_size = mapper['physical blocks']
+        self.physical_used = mapper['data blocks used']
+        self.overhead_used = mapper['overhead blocks used']
+        self.logical_size = mapper['logical blocks']
+        self.logical_used = mapper['logical blocks used']
+
+    def get_slab_size(self, vol):
+        """str: slab size"""
+        self.__read_key_sizes__(vol)
```
Comment (xiangce) [inline insights/combiners/vdo_status_calculation.py]: When getting the `slab_size`, all other attributes will be calculated which is redundant compute.

You can just `self.data['VDOs'][vol]['Slab size']` directly.

The same comment to the following `get_*`.

File: insights/parsers/vdo_status.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    def __get_dev_mapper__(self, vol):
+        try:
+            dm_path = ('/dev/mapper/%s' % vol)
+            return self.data['VDOs'][vol]['VDO statistics'][dm_path]
+        except:
+            raise ParseException("couldn't parse yaml")
+
+    def get_all_volumns(self):
+        """list: vdo volumns"""
+        return self.volumns
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: `return sorted(self.data['VDOs'].keys()) if 'VDOs' in self.data else []`

If OK to you, you could also change the method to a property which has the same usage to attribute.

```python
@property
def volumns(self)
  return sorted(self.data['VDOs'].keys()) if 'VDOs' in self.data else []
```

File: insights/parsers/vdo_status.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+from __future__ import division
+from insights import parser
+from insights import YAMLParser
+from insights.specs import Specs
+from insights.parsers import ParseException
+
+
+@parser(Specs.vdo_status)
+class VDOStatus(YAMLParser):
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: In the title, you have used `VdoStatus`, please uniform it in both the doc and code.

File: insights/parsers/vdo_status.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        1047868
+        >>> vdo.get_logical_used('vdo1')
+        0
+        >>> vdo.get_overhead_used('vdo1')
+        787140
+
+    Attributes:
+
+        data (dict): the result parsed of 'vdo status'
+
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: Do you add these blank lines intentionally? Is it OK to remove them?

File: insights/parsers/vdo_status.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        volumns (list): The list the vdo volumns involved
+
+        dmapper (dict): The dict of device mapper vdo under 'VDO statistics'
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: I haven't seen this attribute is defined in the class's implementation.

File: insights/parsers/vdo_status.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+        vdo slab size
+        vdo volumns
+        vdo data blocks used
+        vdo logical blocks used
+        vdo overhead blocks used
+        vdo physical blocks
+        vdo savings ratio
+        vdo physical used pct
+        vdo logical free savdings ratio pct
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: `savdings` ?

File: insights/parsers/vdo_status.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+
+@parser(Specs.vdo_status)
+class VDOStatus(YAMLParser):
+    """
+    Class for parsing ``vdo status`` command output.
+
+    This class includes GETTING following information of ``vdo status``::
+
+        vdo slab size
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: I believe you mean the `slab size` of specified volumn, but not the slab size of the `vdo` that I can get from the literal.

File: insights/parsers/vdo_status.py [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+        vdo volumns
+        vdo data blocks used
```
Comment (xiangce) [inline insights/parsers/vdo_status.py]: The same comment as above to the following lines.

**Pull request merged**